eventlet/green/ssl.py

```diff
@@ -40,59 +53,56 @@ class GreenSSLSocket(_original_sslsocket):
     settimeout(), and to close/reopen the connection when a timeout
     occurs at an unexpected juncture in the code.
     """
+    def __new__(cls, sock=None, keyfile=None, certfile=None,
+                server_side=False, cert_reqs=CERT_NONE,
+                ssl_version=PROTOCOL_SSLv23, ca_certs=None,
+                do_handshake_on_connect=True, *args, **kw):
+        if sys.version_info < (3, 7):
+            return super(GreenSSLSocket, cls).__new__(cls)
```

Comment:
I'm sure you need to pass cls to __new__. See this example:
```
In [151]: SSLSocket.__new__()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-151-fc85ecb27b2b> in <module>()
----> 1 SSLSocket.__new__()

TypeError: _socket.socket.__new__(): not enough arguments

In [152]: SSLSocket.__new__(SSLSocket)
Out[152]: <ssl.SSLSocket fd=-1, family=AddressFamily.AF_UNSPEC, type=0, proto=0>
```

sock, keyfile should not be sent to __new__ because they are not used. You can do that, but there's actually no effect. It will confuse the maintainer. There are tons of great post explaining the difference between __init__ and __new__. They worth a good read if you have no idea how do __new__ and __init__ work, but feel free to let me know if you want my own understanding.

Reply:
> I'm sure you need to pass cls to `__new__`.

I'm sure of that too. What I'm suggesting is that I believe `super(GreenSSLSocket, cls).__new__()` binds `cls` to the callable returned by `super()` such that you shouldn't need to pass it *again* in the argument list.

But you're saying that this `__new__()` ignores its argument list -- which probably makes it harmless to pass `cls` redundantly.

Reply:
I read through the Googled results again and this is the recommended way of calling parent class's __new__. What's your suggested way of writing the statement? @nat-goodspeed